### ISSUE TITLE
CSS Class Namespacing

### ISSUE THREAD

**Issue by TheWebTech** [TheWebTech/CrankShaft#15]
Our classes should probably be prefixed to ensure we don't conflict with commonly used class names.

Example:
All class names in crankshaft could be prefixed with`.cs-` 

I think that feels long and would get tedious, but would be less likely to conflict. Anyone else got suggestions?

**Comment by adesignl** [comment]
I say that they should follow a bem style configutation

** tabs module **
```
.tab-module {}
.tab-module__tab {}
.tab-module__tab--active {}
.tab-module__panel {}
.tab-module__panel--active {}
```

**Comment by adesignl** [comment]
name space for specific module not the framework

**Comment by TheWebTech** [comment]
I agree with BEM. you may be right about namespacing not being super important, bootstrap's approach is

"We give you these base elements, you decide to use the same class names as us, that's your problem deal with it."

I do think namespacing the modules like you said is an absolute must.

What I want to do is get a guideline doc/wiki page or documentation site page going with best practices for building custom modules. Not all of them would necessarily be best practices for the framework itself some might simply be best practices for module development in general. Like namespacing the module classes and css. BEM usage.
